traffic_ops/traffic_ops_golang/cachegroup/cachegroups.go

```diff
@@ -92,29 +92,47 @@ func (cg *TOCacheGroup) SetID(i int) {
 	cg.ID = &i
 }
 
-// checks if a cachegroup with the given ID is in use as a parent or secondary parent.
-func isUsedByChildCache(tx *sqlx.Tx, ID int) (bool, error) {
-	pQuery := "SELECT count(*) from cachegroup WHERE parent_cachegroup_id=$1"
-	sQuery := "SELECT count(*) from cachegroup WHERE secondary_parent_cachegroup_id=$1"
-	count := 0
-
-	err := tx.QueryRow(pQuery, ID).Scan(&count)
+// Is the cachegroup being used?
+func isUsed(tx *sqlx.Tx, ID int) (bool, error) {
+
+	var servers int
+	var pid int
+	var pid2 int
+	var asns int
+
+	//Counts all the places you might find a cachegroup
+	query := `select
```

Comment:
Don't subqueries get expensive, especially if each one does `count(*)`?
I'm currently re-writing the query so that the parent and secondary parent are from different joins (and therefore count correctly). I can add column aliases so running the query locally will be readable, but I think it is readable enough using joins instead of subqueries. Will you reject the PR for this?